snpdb/models/models_variant.py

```diff
@@ -254,21 +254,6 @@ class VariantCoordinate(FormerTuple):
     ref: str
     alt: str
 
-    def __init__(self, chrom, start, end: int = None, ref: str = None, alt: str = None):
```

Comment:
Though TypeHints might be able to pickup on some bad usage of VariantCoordinate, it's common enough that maybe validation in a __post_init__(): method might be a good start into "fail early" design.

Reply:
Added simple type checking (not sure what else you want - or whether we can do this automatically off type hints

Reply:
Re off type hints, found this seems designed specifically to ensure a dataclass has all the correct values
https://pypi.org/project/enforce-typing/

(we can introduce that as work after the merge to stop holding the merge back if you'd prefer)